otx/algorithms/anomaly/adapters/anomalib/callbacks/inference.py

```diff
@@ -49,49 +52,118 @@ def on_predict_epoch_end(self, _trainer: pl.Trainer, pl_module: AnomalyModule, o
+                    numpy=anomaly_map_to_color_map(anomaly_map.squeeze().numpy(), normalize=False),
                 )
             )
-            log_string = (
-                f"\n\tThreshold: {pl_module.image_threshold.value.item():.3f},"
-                f" Assigned Label '{label.name}', {pred_score:.3f}"
+
+    def _process_classification_predictions(self, pred_labels: Tensor, pred_scores: Tensor):
+        """Add classification predictions to the dataset items.
+
+        Args:
+            pred_labels (Tensor): Predicted image labels.
+            pred_scores (Tensor): Predicted image-level anomaly scores.
+        """
+        for dataset_item, pred_label, pred_score in zip(self.otx_dataset, pred_labels, pred_scores):
+            # get label
+            label = self.anomalous_label if pred_label else self.normal_label
+            probability = pred_score if pred_label else 1 - pred_score
+            # update dataset item
+            dataset_item.append_labels([ScoredLabel(label=label, probability=float(probability))])
+
+    def _process_detection_predictions(
+        self,
+        pred_boxes: List[Tensor],
+        box_scores: List[Tensor],
+        box_labels: List[Tensor],
+        pred_scores: Tensor,
+        image_size: torch.Size,
+    ):
+        """Add detection predictions to the dataset items.
+
+        Args:
+            pred_boxes (List[Tensor]): Predicted bounding box locations.
+            box_scores (List[Tensor]): Predicted anomaly scores for the bounding boxes.
+            box_labels (List[Tensor]): Predicted labels for the bounding boxes.
+            pred_scores (Tensor): Predicted image-level anomaly scores.
+            image_size: (torch.Size): Image size of the original images.
+        """
+        # TODO; refactor Ignore too many locals
+        # pylint: disable=too-many-locals
+        height, width = image_size
+        for dataset_item, im_boxes, im_box_scores, im_box_labels, pred_score in zip(
+            self.otx_dataset, pred_boxes, box_scores, box_labels, pred_scores
+        ):
+            # generate annotations
+            annotations: List[Annotation] = []
+            for box, score, label in zip(im_boxes, im_box_scores, im_box_labels):
+                if box[0] >= box[2] or box[1] >= box[3]:  # discard 1-pixel boxes
+                    continue
+                shape = Rectangle(
+                    x1=box[0].item() / width,
+                    y1=box[1].item() / height,
+                    x2=box[2].item() / width,
+                    y2=box[3].item() / height,
+                )
+                label = self.label_map[label.item()]
+                probability = score.item()
+                annotations.append(Annotation(shape=shape, labels=[ScoredLabel(label=label, probability=probability)]))
+            # get label
+            label = self.normal_label if len(annotations) == 0 else self.anomalous_label
```

Comment:
A suggestion by `refurb`:
```suggestion
            label = self.anomalous_label if annotation else self.normal_label
```